embassy-nrf/src/twim.rs

```diff
@@ -270,10 +272,6 @@ impl<'d, T: Instance> Twim<'d, T> {
 
         self.read_errorsrc()?;
 
-        if r.txd.amount.read().bits() != buffer.len() as u32 {
-            return Err(Error::Transmit);
-        }
-
```

Comment:
How come this disappeared?

Reply:
Hmm, I thought it couldn't happen but it seems it can: https://github.com/NordicSemiconductor/nrfx/blob/master/drivers/src/nrfx_twim.c#L167

I'll check what's up and fix

Reply:
These error checks are now back.